### PR TITLE
Add coverage checks for Python 3.7-3.10

### PR DESCRIPTION
**Pull request by knsmith**
Addresses https://github.com/SupertechLabs/general-superstaq/issues/165 for cirq-superstaq

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -88,15 +88,16 @@ jobs:
 
   coverage:
     name: Pytest and Coverage check
-    env:
-      PRIVATE_REPOS_KEY: ${{ secrets.PRIVATE_REPOS_KEY }}
+    strategy:
+      matrix:
+        python-version: ['3.7', '3.8', '3.9', '3.10']
```
Comment (vtomole) [inline .github/workflows/ci.yml:93]:
```suggestion
        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
```

Comment (vtomole) [reply .github/workflows/ci.yml:93]: nvm the dev version of  css uses qiskit.

Comment (knsmith) [reply .github/workflows/ci.yml:95]: added back 
`env:`
      `PRIVATE_REPOS_KEY: ${{ secrets.PRIVATE_REPOS_KEY }}`

**Comment by knsmith** [comment]
@vtomole ready for another review!

**Review by vtomole — APPROVED** [review]
LGTM!

**Pull request merged**